packages/react-magma-dom/src/components/CharacterCounter/CharacterCounter.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+import { FormFieldContainerProps } from '../FormFieldContainer';
+import { ThemeContext } from '../../theme/ThemeContext';
+import { ThemeInterface } from '../../theme/magma';
+import { I18nContext } from '../../i18n';
+import { InputMessage } from '../Input/InputMessage';
+
+export interface CharacterCounterProps
+  extends Omit<FormFieldContainerProps, 'fieldId'>,
+    React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * Total number of characters in an input.
+   */
+  inputLength?: number;
+  /**
+   * Sets the maximum amount of characters allowed.
+   */
+  maxLength?: number;
+  testId?: string;
+  /**
+   * @internal
+   */
+  theme?: ThemeInterface;
```

Comment:
Theme does not need to be a prop of CharacterCounter (you don't pass theme to it when you call it in FormFieldContainer)